transifex/utils/common.py

```diff
@@ -0,0 +1,283 @@
+import logging
+import os
+import re
+import subprocess
+import time
+from contextlib import contextmanager
+from logging.config import dictConfig
+from urllib.parse import urlparse
+
+import yaml
+from github import Github, GithubException
+
+# Configure logging.
+dictConfig({
+    'version': 1,
+    'disable_existing_loggers': False,
+    'formatters': {
+        'standard': {
+            'format': '%(asctime)s %(levelname)s %(process)d [%(filename)s:%(lineno)d] - %(message)s',
+        },
+    },
+    'handlers': {
+        'console': {
+            'level': 'INFO',
+            'class': 'logging.StreamHandler',
+            'formatter': 'standard',
+        },
+    },
+    'loggers': {
+        '': {
+            'handlers': ['console'],
+            'level': 'DEBUG',
+            'propagate': False
+        },
+    },
+})
+logger = logging.getLogger()
+
+
+@contextmanager
+def cd(repo):
+    """Utility for changing into and out of a repo."""
+    initial_directory = os.getcwd()
+    os.chdir(repo.name)
+
+    # Exception handler ensures that we always change back to the
+    # initial directory, regardless of how control is returned
+    # (e.g., an exception is raised while changed into the new directory).
+    try:
+        yield
+    finally:
+        os.chdir(initial_directory)
+
+
+@contextmanager
+def repo_context(*args, **kwargs):
+    """Utility for cloning a repo, cd'ing to it, creating a working branch, doing some work, and then cleaning
+       everything up.
+    """
+    repo = Repo(*args, **kwargs)
+    try:
+        repo.clone()
+        with cd(repo):
+            repo.branch()
+            yield repo
+    finally:
+        repo.cleanup()
```

Comment:
I would add a note about this being a good idea because it cleans up the (Jenkins) workspace even in the event of a failure, allowing the script to be rerun without issue.